src/wallet/load.cpp

```diff
@@ -27,9 +29,9 @@ bool VerifyWallets(WalletContext& context)
 
     if (args.IsArgSet("-walletdir")) {
         fs::path wallet_dir = fs::PathFromString(args.GetArg("-walletdir", ""));
-        boost::system::error_code error;
+        std::error_code error;
         // The canonical path cleans the path, preventing >1 Berkeley environment instances for the same directory
-        fs::path canonical_wallet_dir = fs::canonical(wallet_dir, error).remove_trailing_separator();
+        fs::path canonical_wallet_dir = fs::canonical(wallet_dir, error);
```

Comment:
In commit "refactor: replace boost::filesystem with std::filesystem" (41d7166c8a598b604aad6c6b1d88ad46e23be247)

Note: std canonical function already strips trailing slashes, so this should not be a change in behavior